### PR TITLE
uname and common functions

### PR DESCRIPTION
**Pull request by Kle0s**
- added `suname` implementation.
- created a few common function for the use of other commands:
    - `handle_errors`: prints coreutils-like error messages.
    - `write_out`: basic write function to **stdout**.
    - `write_err`: basic write function to **stderr**.
    - `exit`: a basic exit function.
- edited `sarch` implementation to fit new model.

### CODE REVIEW

File: src/common.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    An implementation of basic error handling to print the correct error message for each case.
+    """
+    command_name = _get_command_name()
+    if sys.argv[1].startswith(FULL_ARG_PREFIX):
+        write_out(f"{command_name}: unrecognized option '{sys.argv[1]}'")
+    elif sys.argv[1].startswith("-") and 1 < len(sys.argv[1]):
+        write_out(f"{command_name}: invalid option -- '{sys.argv[1][1:]}'")
+    else:
+        write_out(f"{command_name}: extra operand '{sys.argv[1]}'")
+    write_out(f"Try '{command_name} --help' for more information.")
```
Comment (honestly-nuts) [inline src/common.py:59]: Why is sys.argv[1] hardcoded?

**Review by honestly-nuts — COMMENTED** [review]
I like the code over all but I suggest removing the hardcoded sys.argv[1]'s (in handla_errors() ) and doing something more dynamic.

File: src/common.py (hunk 1 above)
Comment (Kle0s) [reply src/common.py:59]: this is based on the assumption the argument list is given there by the shell. 
what do you suggest?

Comment (Kle0s) [reply src/common.py:59]: I'll add an `argument_list` parameter to the function

**Pull request merged**